bokeh/document.py

```diff
@@ -298,6 +300,12 @@ def __init__(self, **kwargs):
         self._callbacks = {}
         self._session_callbacks = {}
         self._session_context = None
+        self._modules = []
+
+    def __del__(self):
+        for module in self._modules:
+            if module.__name__ in sys.modules:
+                del sys.modules[module.__name__]
```

Comment:
Tornado is a single python process. Every session served by a given bokeh server has its  own document. If the app uses a custom module, wouldn't the custom models be shared amongst all of the documents for all of the sessions in the one `sys.modules` for the single Tornado process? Maybe I am thinking of something else or missing something obvious.